cli/tests/integration/run_tests.rs

```diff
@@ -1179,6 +1166,143 @@ fn lock_deno_json_package_json_deps() {
+  package_json.write_json(&json!({
+    "workspaces": ["package-a"],
+    "dependencies": {
+      "@denotest/cjs-default-export": "1"
+    }
+  }));
+  // main.ts
+  let main_ts = temp_dir.join("main.ts");
+  main_ts.write("import '@denotest/cjs-default-export';");
+
+  // package-a/package.json
+  let a_package = temp_dir.join("package-a");
+  a_package.create_dir_all();
+  let a_package_json = a_package.join("package.json");
+  a_package_json.write_json(&json!({
+    "dependencies": {
+      "@denotest/esm-basic": "1"
+    }
+  }));
+  // package-a/main.ts
+  let main_ts = a_package.join("main.ts");
+  main_ts.write("import '@denotest/esm-basic';");
+  context
+    .new_command()
+    .args("run package-a/main.ts")
+    .run()
+    .skip_output_check();
+  let lockfile = temp_dir.join("deno.lock");
+  let esm_basic_integrity =
+    get_lockfile_npm_package_integrity(&lockfile, "@denotest/esm-basic@1.0.0");
+
+  // no "workspace" because deno isn't smart enough to figure this out yet
+  // since it discovered the package.json in a folder different from the lockfile
+  lockfile.assert_matches_json(json!({
+    "version": "3",
+    "packages": {
+      "specifiers": {
+        "npm:@denotest/esm-basic@1": "npm:@denotest/esm-basic@1.0.0"
+      },
+      "npm": {
+        "@denotest/esm-basic@1.0.0": {
+          "integrity": esm_basic_integrity,
+          "dependencies": {}
+        }
+      }
+    },
+    "remote": {},
+  }));
+
+  // run a command that causes discovery of the root package.json beside the lockfile
+  context
+    .new_command()
+    .args("run main.ts")
+    .run()
+    .skip_output_check();
+  // now we should see the dependencies
+  let cjs_default_export_integrity = get_lockfile_npm_package_integrity(
+    &lockfile,
+    "@denotest/cjs-default-export@1.0.0",
+  );
+  let expected_lockfile = json!({
+    "version": "3",
+    "packages": {
+      "specifiers": {
+        "npm:@denotest/cjs-default-export@1": "npm:@denotest/cjs-default-export@1.0.0",
+        "npm:@denotest/esm-basic@1": "npm:@denotest/esm-basic@1.0.0"
+      },
+      "npm": {
+        "@denotest/cjs-default-export@1.0.0": {
+          "integrity": cjs_default_export_integrity,
+          "dependencies": {}
+        },
+        "@denotest/esm-basic@1.0.0": {
+          "integrity": esm_basic_integrity,
+          "dependencies": {}
+        }
+      }
+    },
+    "remote": {},
+    "workspace": {
+      "packageJson": {
+        "dependencies": [
+          "npm:@denotest/cjs-default-export@1"
+        ]
+      }
+    }
+  });
+  lockfile.assert_matches_json(expected_lockfile.clone());
+
+  // now run the command again in the package with the nested package.json
+  context
+    .new_command()
+    .args("run package-a/main.ts")
+    .run()
+    .skip_output_check();
+  // the lockfile should stay the same as the above because the package.json
+  // was found in a different directory
+  lockfile.assert_matches_json(expected_lockfile.clone());
```

Comment:
Previously this would flip flop back and forth between the packages from `package.json` and `packages-a/package.json`. Now it just stays being the ones in `package.json`. In the future, we should get deno to understand npm workspaces beyond just for byonm and then it could include all the members in the lockfile (luckily the lockfile code already has support for this... it's just getting it into the CLI sometime).